File: Programmers/Lv1/Lv1_콜라츠추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution53(int num) {
	int answer = 0;
	long n = num;
	while (n != 1) {
		if (n % 2 == 0) {
			n /= 2;
			answer++;
		}
		else {
			n *= 3;
			n += 1;
			answer++;
		}
		if (answer > 500) {
			answer = -1;
			break;
		}
	}
	return answer;
}

File: Programmers/Lv1/Lv1_콜라츠추측.py
def solution(num):
    answer = 0
    while num!=1:
        if answer==500 :
            break
        if num%2==0:
            num = num/2;
        else:
            num = num*3 + 1
        answer += 1
    if num==1:
        return answer
    else:
        return -1